src/hooks/navigation.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { createContext } from 'react';
+import { useNavigation as useRNNavigation, useRoute } from '@react-navigation/native';
+
+/* global BigInt */
+
+// This prefix helps identify serialized BigInt values
+const BIG_INT_PREFIX = '__BIGINT__:';
+
+// Recursive serialization/deserialization functions
+function serialize(data) {
```

Comment:
Do you think it makes sense to return a clone (from lodash) of the data in both the serialize and deserialize? To avoid returning a reference somewhere.